resources/sdk/clisdkclient/extensions/cmd/profiles/createprofile.go

```diff
@@ -77,35 +104,100 @@ func requestUserInput() config.Configuration {
 		environment = "mypurecloud.com"
 	}
 
-	fmt.Printf("Please provide either an Access Token, Client Credentials (Client ID and Client Secret) or both\n")
-	fmt.Printf("Note: If you provide an Access Token, this will be used over Client Credentials\n")
-	for {
-		fmt.Printf("Access Token: ")
-		fmt.Scanln(&accessToken)
-		fmt.Printf("Client ID: ")
-		fmt.Scanln(&clientID)
-		fmt.Printf("Client Secret: ")
-		fmt.Scanln(&clientSecret)
-		// users should only be allowed to continue if they provide an access token
-		if len(strings.TrimSpace(accessToken)) != 0 && len(strings.TrimSpace(clientID)) == 0 && len(strings.TrimSpace(clientSecret)) == 0 {
-			clientID = ""
-			clientSecret = ""
-			break
-			// or if they provide client credentials
-		} else if len(strings.TrimSpace(accessToken)) == 0 && len(strings.TrimSpace(clientID)) != 0 && len(strings.TrimSpace(clientSecret)) != 0 {
-			accessToken = ""
-			break
-			// or both
-		} else if len(strings.TrimSpace(accessToken)) != 0 && len(strings.TrimSpace(clientID)) != 0 && len(strings.TrimSpace(clientSecret)) != 0 {
+	fmt.Print("Note: If you provide an access token, this will take precedence over any authorization grant type.\n")
+	fmt.Print("Access Token (Optional): ")
+	fmt.Scanln(&accessToken)
+
+	for true {
+		fmt.Print("Select your authorization grant type.\n")
+		fmt.Print("\t0. None\n\t1. Client Credentials\n\t2. Implicit Grant\nGrant Type: ")
+		fmt.Scanln(&grantType)
+
+		if accessToken == "" && grantType == None {
+			fmt.Print("If you have not provided an access token, you must select a grant type.\n")
+			continue
+		}
+		if isValidGrantType(grantType) {
 			break
-			// otherwise
+		}
+	}
+
+	clientID, clientSecret = requestClientCreds(accessToken, grantType)
+
+	if grantType == ImplicitGrant {
+		for true {
+			fmt.Print("Would you like to use a secure HTTP connection? [Y/N]: ")
+			fmt.Scanln(&authChoice)
+			if strings.ToUpper(authChoice) == "Y" {
+				secureLoginEnabled = true
+				break
+			} else if strings.ToUpper(authChoice) == "N" {
+				secureLoginEnabled = false
+				break
+			}
+		}
+		redirectURI = requestRedirectURI(secureLoginEnabled)
+		fmt.Printf("Redirect URI: %s\n", redirectURI)
+	}
+
+	return constructConfig(name, environment, clientID, clientSecret, redirectURI, secureLoginEnabled, accessToken)
+}
+
+func requestClientCreds(accessToken string, grantType GrantType) (string, string) {
+	id := ""
+	secret := ""
+
+	if grantType == ClientCredentials {
+		if accessToken != "" {
+			fmt.Print("Client ID: ")
+			fmt.Scanln(&id)
+
+			fmt.Print("Client Secret: ")
+			fmt.Scanln(&secret)
```

Comment:
Do you mean to avoid it showing up on the terminal as it's entered? Something like this question? https://stackoverflow.com/questions/2137357/getpasswd-functionality-in-go
If so looks like there's some good solutions

Reply:
I read the code without first consuming a sufficient amount of coffee; you're not logging the value like I said. Hiding the input value would be excellent if it's not too difficult to implement. Using a user input library like that probably has some other fringe benefits as well.